src/Microsoft.PowerShell.Commands.Management/commands/management/ContentCommandBase.cs

```diff
@@ -137,99 +137,106 @@ internal void WriteContentObject(object content, long readCount, PathInfo pathIn
+            [PSExtensionMember]
+            public string PSParentPath => _parentPath;
 
-                    // Get the child name
+            [PSExtensionMember]
+            public string PSPath => LocationGlobber.GetProviderQualifiedPath(_pathInfo.Path, _pathInfo.Provider);
 
-                    string childName = SessionState.Path.ParseChildName(pathInfo.Path, context);
-                    note = new PSNoteProperty("PSChildName", childName);
-                    result.Properties.Add(note, true);
-                    tracer.WriteLine("Attaching {0} = {1}", "PSChildName", childName);
-                    _currentContentItem.ChildName = childName;
-                }
-                catch (NotSupportedException)
-                {
-                    // Ignore. The object just won't have ParentPath or ChildName set.
-                }
+            [PSExtensionMember]
+            public string PSChildName => _childName;
 
-                // PSDriveInfo
+            [PSExtensionMember]
+            public long ReadCount => _readCount;
 
-                if (pathInfo.Drive != null)
+            [PSExtensionMember]
+            public ProviderInfo PSProvider => _pathInfo.Provider;
+
+            T IPSObjectExtendedMemberInfo.GetMember<T>(MemberNamePredicate predicate)
+            {
+                string GetMatchedPropertyName(MemberNamePredicate pred)
                 {
-                    PSDriveInfo drive = pathInfo.Drive;
-                    note = new PSNoteProperty("PSDrive", drive);
-                    result.Properties.Add(note, true);
-                    tracer.WriteLine("Attaching {0} = {1}", "PSDrive", drive);
-                    _currentContentItem.Drive = drive;
+                    if (pred(nameof(PSParentPath))) return nameof(PSParentPath);
+                    if (pred(nameof(PSPath))) return nameof(PSPath);
+                    if (pred(nameof(PSChildName)))return nameof(PSChildName);
+                    if (pred(nameof(ReadCount)))return nameof(ReadCount);
+                    if (pred(nameof(PSProvider)))return nameof(PSProvider);
+                    return null;
                 }
 
-                // ProviderInfo
-
-                ProviderInfo provider = pathInfo.Provider;
-                note = new PSNoteProperty("PSProvider", provider);
-                result.Properties.Add(note, true);
-                tracer.WriteLine("Attaching {0} = {1}", "PSProvider", provider);
-                _currentContentItem.Provider = provider;
+                var propertyName = GetMatchedPropertyName(predicate);
+                return propertyName == null ? null : DotNetInstanceAdapter.GetDotNetProperty<T>(this, propertyName);
             }
 
-            // Add the ReadCount note
-            note = new PSNoteProperty("ReadCount", readCount);
-            result.Properties.Add(note, true);
+            T IPSObjectExtendedMemberInfo.GetMember<T>(string name) => DotNetInstanceAdapter.GetDotNetProperty<T>(this, name);
```

Comment:
One CI test failed on ReadCount property - perhaps a problem is here.

> because the manifest is not valid. Verify that the manifest file is valid, and then try again.'The property 'readcount' cannot be found on this object. Verify that the property exists.'

Second failed test is "Verify Clear-Item with -WhatIf". We could run commands from the test manually and see which command will wrong.

Reply:
Oh, I have probably borked the original implementation. It works with the feature enabled, but not when it's disabled. Thx @iSazonov

Reply:
@iSazonov You were right - I had missed the ReadCount property when the experimental feature was disabled.